File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    static bool comparePairs(pair<int, int> a, pair<int, int> b) {
        return a.second > b.second;
    }
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        for (int num : nums) {
            m[num]++;
        }

        vector<pair<int, int>> items(m.begin(), m.end());
        sort(items.begin(), items.end(), comparePairs);

        vector<int> to_return;
        for (int i = 0; i < k; i++) {
            to_return.push_back(items[i].first);
        }
        return to_return;
    }
};

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def binary_search(l, r):
            print(l, r)
            i = (l + r) // 2
            print(i)
            
            if (nums[i] == target):
                return i
            if l >= r:
                return -1
            if nums[i] < target:
                return binary_search(i + 1, r)
            elif nums[i] > target:
                return binary_search(l, i-1)
            

        return binary_search(0, len(nums) - 1)